bsd/disk.pyx

```diff
@@ -5,6 +5,28 @@ from fcntl import ioctl
 from . cimport disk_ioctl
 
 
+def get_sectorsize_with_name(disk):
+    """
+    Get sectorsize of disk.
+
+    `disk` must be a string and can be `/dev/da1` or just `da1`
+    """
+    disk = disk.removeprefix('/dev/')
+    with open(f'/dev/{disk}', 'rb') as f:
+        return get_sectorsize_with_file(f)
+
+
+def get_sectorsize_with_file(f):
+    """
+    Get sectorsize of disk.
+
+    `f` must be a python file object.
+    """
+    _buffer = array('B', range(0, 4))
+    ioctl(f.fileno(), disk_ioctl.DIOCGSECTORSIZE, _buffer, 1)
```

Comment:
Should we be checking it's return value to ensure it executed successfully ?

Reply:
This is _EXACTLY_ formatted the same way as the other methods. I don't feel like making this method different than the others.